javascript/javascript1/week3/homework-js1-w3.js

```diff
@@ -114,7 +114,89 @@ logOutSeriesText();
 // NOnoN0nOYes (Note taking app)
 //---------------------------------
 
+// ++++ Save a note ++++
+const notes = [];
+
+function saveNote(content, id) {
+  notes.push({ content: content, id: id });
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+// ++++ Get a note ++++
+function getNote(id) {
+  for (let note of notes) {
+    if (id === note.id) {
+      return note;
+    }
+  }
+}
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+// ++++ Log out notes ++++
+function logOutNotesFormatted() {
+  for (let note of notes) {
+    console.log(
+      `The note with id: ${note.id}, has the following note text: ${note.content}`
+    );
+  }
+}
+
+logOutNotesFormatted(); // should log out the text below
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+// ++++ Unique feature (Delete a note) ++++
+function deleteNote(num) {
+  for (let i = 0; i < notes.length; i++) {
+    if (num === notes[i].id) {
+      notes.splice(i, 1);
+      console.log(`Note with id: ${num} has been deleted.`);
+    }
+  }
+}
+
+deleteNote(1);
+console.log(notes);
 
 //---------------------------------
 // CactusIO-interactive (Smart phone usage app)
 //---------------------------------
+
+// ++++ Adding an activity ++++
+const activities = [];
+function addActivity(date, activity, duration) {
+  activities.push({ date: date, activity: activity, duration: duration });
+}
```

Comment:
This is beyond the scope of our classes but what's really cool is you can actually write a shorthand for when the key and the value is the same namespace.

```
function addActivity(date, activity, duration) {
  activities.push({ date, activity, duration });
}
```

This will do the same as you have done and both create the key with name i.e. `date` as well as passing the value `date`